titan/model.py

```diff
@@ -358,48 +362,112 @@ def agents_interact(self, rel: "ag.Relationship"):
             interaction = self.interactions[interaction_type]
             interaction.interact(self, rel)
 
-    def die_and_replace(self):
+    def exit(self):
+        if self.exits == {}:
+            return
+        for agent in self.pop.all_agents:
+            for strategy in self.params.enter_exit.values():
+                exit = self.params.classes.exit[strategy.exit_class]
+                if exit.ignore_incar and agent.incar.active:
+                    continue
+                if exit.exit_type == "age_out":
+                    # agent ages out of model
+                    if agent.age > exit.age:
+                        self.exits[strategy.exit_class].append(agent)
+                        # agent can only leave once, move to next agent
+                        break
+                elif exit.exit_type == "death":
+                    p = (
+                        prob.get_death_rate(
+                            agent.hiv.active,
+                            agent.hiv.aids,
+                            agent.drug_type,
+                            agent.sex_type,
+                            agent.haart.adherent,
+                            agent.race,
+                            agent.location,
+                            self.params.model.time.steps_per_year,
+                            strategy.exit_class,
+                        )
+                        * self.calibration.mortality
+                    )
+
+                    if self.run_random.random() < p:
+                        # agent dies
+                        self.exits[strategy.exit_class].append(agent)
+                        break
+                elif exit.exit_type == "drop_out":
+                    p = (
+                        agent.location.params.demographics[agent.race]
+                        .sex_type[agent.sex_type]
+                        .drug_type[agent.drug_type]
+                        .exit[strategy.exit_class]
+                        .prob
+                    )
+                    if self.run_random.random() < p:
+                        # agent leaves study pop
+                        self.exits[strategy.exit_class].append(agent)
+                        break
```

Comment:
```suggestion
                        break

```

Reply:
also, this is the last statement, do we need the break?

Reply:
Nope!